Engine/source/T3D/fx/lightning.h

```diff
@@ -86,6 +92,22 @@ class LightningData : public GameBaseData
 
    DECLARE_CONOBJECT(LightningData);
    static void initPersistFields();
+
+   SFXProfile* getThunderProfile(U32 id)
+   {
+      if (mThunderSoundAsset[id] != NULL)
+         return mThunderSoundAsset[id]->getSfxProfile();
+      else
+         return NULL;
+   }
+
+   SFXProfile* getSFXProfile() {
```

Comment:
Uh oh. `One true brace` coding style leaked here...

Also it is better coding style for functions that return values to always have a non-conditional return statement at the end of the function, instead of nested under an else, like so:

```cpp
SFXProfile* getSFXProfile()
{
   if (mStrikeSoundAsset.notNull())
      return mStrikeSoundAsset->getSfxProfile();

   return NULL;
}
```

That way the intent is more clear. If mStrikeSoundAsset is not NULL return it. At end of function it just returns NULL as in failure to satisfy request, operation failed, etc..

Reply:
copied from the way it was done before

Reply:
https://github.com/TorqueGameEngines/Torque3D/blob/Preview4_0/Engine/source/T3D/fx/precipitation.h#L76